.github/get-packages.py

```diff
@@ -37,33 +32,24 @@ def _parse_args() -> str:
 
 
 def _main(git_base_ref: str) -> None:
-    packages = _get_changed_packages(git_base_ref=git_base_ref)
-    line = f'packages={json.dumps(packages)}'
-    print(line)
-    with pathlib.Path(os.environ['GITHUB_OUTPUT']).open('a') as f:
-        print(line, file=f)
-
-
-def _get_changed_packages(git_base_ref: str) -> list[str]:
-    all_packages = [*_get_packages('.', exclude='interfaces'), *_get_packages('interfaces')]
+    cmd = ['.scripts/ls.py', 'packages']
     if not git_base_ref:
         print('Using all packages because no git base ref was provided:')
-        return all_packages
-    cmd = ['git', 'diff', '--name-only', f'origin/{git_base_ref}']
-    output = subprocess.check_output(cmd, text=True)
-    diff = [p.split('/') for p in output.split('\n')]
-    changes = {*(p[0] for p in diff), *('/'.join(p[:2]) for p in diff)}
-    if global_changes := sorted(changes.intersection(_GLOBAL_FILES)):
+    elif global_changes := _get_global_changes(git_base_ref):
         print(f'Using all packages because global files were changed: {global_changes}')
-        return all_packages
-    print(f'Using packages that are changed compared to {git_base_ref}:')
-    return sorted(changes.intersection(all_packages))
+    else:
+        cmd.append(git_base_ref)
+    packages = subprocess.check_output(cmd, text=True).strip()
+    line = f'packages={packages}'
+    print(line)
+    with pathlib.Path(os.environ['GITHUB_OUTPUT']).open('a') as f:
+        print(line, file=f)
 
 
-def _get_packages(root: pathlib.Path | str, exclude: str | None = None) -> list[str]:
-    root = pathlib.Path(root)
-    paths = [root / '.package', root / '.example', root / '.tutorial', *root.glob(r'[a-z]*')]
-    return sorted(str(path) for path in paths if path.is_dir() and path.name != exclude)
+def _get_global_changes(git_base_ref: str) -> list[str]:
+    cmd = ['git', 'diff', '--name-only', git_base_ref]
+    changes = subprocess.check_output(cmd, text=True).strip().splitlines()
+    return sorted(_GLOBAL_FILES.intersection(c.split('/')[0] for c in changes))
```

Comment:
It could be just me, but it took me a while to figure this line out because I kept misreading where the parentheses were. What about making it more lines?

```suggestion
    changed_roots = {c.split('/')[0] for c in changes}
    changed_global_files = _GLOBAL_FILES.intersection(changed_roots)
    return sorted(changed_global_files)
```

Or even a two-line version that merged the last of those two in my example.

Reply:
Good suggestion, I'll switch to a two line version.